addon/helpers/render-markdown.js

```diff
@@ -1,36 +1,64 @@
+import Ember from 'ember';
+
 import { isEmpty } from '@ember/utils';
 import { helper } from '@ember/component/helper';
 import { htmlSafe } from '@ember/string';
+
 import markdownit from 'markdown-it';
 import attrs from 'markdown-it-attrs';
 
-function targetLinks(html) {
-  let origin = window.location.origin;
+const {
+  Handlebars: {
+    Utils: {
+      escapeExpression,
+    },
+  },
+} = Ember;
+
+// SEE: https://github.com/markdown-it/markdown-it/blob/master/docs/architecture.md#renderer
+
+const md = markdownit({ html: true }).use(attrs);
 
-  let parser = new DOMParser();
-  let doc = parser.parseFromString(html, 'text/html');
-  let nodes = doc.querySelectorAll(`a[href^='mailto'], a[href^='http']:not([href^='${origin}'])`);
+// Remember old renderer, if overridden, or proxy to default renderer
+const defaultRender = md.renderer.rules.link_open || function(tokens, idx, options, _env, self) {
+  return self.renderToken(tokens, idx, options);
+};
 
-  for (let i = 0; i < nodes.length; i++) {
-    let node = nodes[i];
+// Add target and rel to links when anchor tag is opened
+md.renderer.rules.link_open = function(tokens, idx, options, env, self) {
+  tokens[idx].attrPush(['target', '_blank']);
+  tokens[idx].attrPush(['rel', 'noopener noreferrer']);
 
-    node.setAttribute('target', '_blank');
-    node.setAttribute('rel', 'noopener noreferrer');
+  // pass token to default renderer.
+  return defaultRender(tokens, idx, options, env, self);
+};
+
+export function parseDefinitions(html) {
+  // SEE: https://regex101.com/r/3Dpban/4
+  const DEFINITIONS_EXPR = /\[definition:[\s+]?(?<definition>.*?)\](?<term>.*?)\[\/definition\]/g
```

Comment:
can you link here to where this feature is documented in notion so one can see a usage example 
maybe even add one inline here?

Reply:
The regex101.com link has examples, but I can put one inline here as well.